Warnings for implicit declarations in XPCE

Compiling from sources on Ubuntu 16.10 I get the following warnings that I haven't seen before:

```
QA Notice: Package triggers severe warnings which indicate that it
           may exhibit random runtime failures.
unx/process.c:586:7: warning: implicit declaration of function ‘grantpt’ [-Wimplicit-function-declaration]
unx/process.c:587:7: warning: implicit declaration of function ‘unlockpt’ [-Wimplicit-function-declaration]
unx/process.c:588:15: warning: implicit declaration of function ‘ptsname’ [-Wimplicit-function-declaration]

QA Notice: Package triggers severe warnings which indicate that it
           may exhibit random runtime failures.
unx/process.c:588:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
unx/process.c:637:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
```

There is indeed a few lines in `packages/xpce/src/unx/process.c` that mention the absence of these declarations:

```c
#if HAVE_STROPTS_H && HAVE_GRANTPT	/* Solaris */
#define USE_GRANTPT 1
#include <stropts.h>
#ifdef __mips__				/* These prototypes are nowhere!? */
extern int   grantpt(int filedes);
extern char *ptsname(int fildes);
extern int   unlockpt(int fildes);
#endif
#endif
```

--- a/src/unx/process.c
+++ b/src/unx/process.c
@@ -33,7 +33,7 @@
 */
 
 #if defined(__linux__) || defined(__GLIBC__) || defined(__GNU__)
-#define _XOPEN_SOURCE			/* GNU glibc grantpt() prototypes */
+#define _XOPEN_SOURCE 500 /* GNU glibc grantpt() prototypes */
 #endif
 #include <h/kernel.h>
 

